lib/tram/policy/error.rb

```diff
@@ -0,0 +1,82 @@
+require "i18n"
+require 'core_ext/string'
+require 'core_ext/hash'
+
+module Tram
+  class Policy
+    class Error
+      attr_reader :policy, :tags, :message
+
+      def initialize(policy, message, tags)
+        @policy = policy
+        @tags = tags.symbolize_keys
+        @initial_message = message
+        @message = generate_message(message)
+      end
+
+      # Return the message with tags info added
+      #
+      #   error = Tram::Policy::Error.new(policy, "Title is empty", field: "title", level: "error")
+      #   error.full_message # => "Title is empty: {:field=>\"title\", :level=>\"error\"}"
+      def full_message
+        "#{@message}: #{@tags}"
+      end
+
+      # Translates an error message in its default scope
+      #
+      # Say you have class Article::ReadinessPolicy < Tram::Policy; end
+      # and you wanted the translation for the :blank error message for
+      # the title attribute, it looks for this translation: article/readiness_policy.empty
+      def generate_message(message)
+        if message.is_a?(String)
+          message
+        elsif message.is_a?(Symbol)
+          I18n.t(message, @tags.merge(scope: i18n_scope, default: "Error translation for missed text"))
+        else
+          raise ArgumentError.new("Only strings or symbols are allowed")
+        end
+      end
+
+      # Return an array of missed translations to every available locale
+      #
+      #   Tram::Policy::Error.new article_policy, :empty_title, field: 'title'
+      #   error.missed_translations # => ['en.article.empty_title', 'ru.article.empty_title']
+      def missed_translations
```

Comment:
I meant https://www.relishapp.com/rspec/rspec-expectations/v/3-5/docs/custom-matchers
(I warmly recommend you reading the whole RSpec documentation on relish -- it's worth it)